cpp/src/Slice/Parser.h

```diff
@@ -900,22 +877,22 @@ class Sequence : public virtual Constructed
 // Dictionary
 // ----------------------------------------------------------------------
 
-class Dictionary : public virtual Constructed
+class Dictionary : public virtual Contained, public virtual Type
 {
 public:
 
     TypePtr keyType() const;
     TypePtr valueType() const;
     StringList keyMetadata() const;
     StringList valueMetadata() const;
+    std::string typeId() const override;
```

Comment:
Do we need all these type IDs?

Ice itself only needs type IDs for classes (for marshaling), exceptions (for marshaling) and interfaces (for operations like ice_isA and ice_id/ice_ids), so maybe we could move `typeId` to ClassDecl, InterfaceDecl and Exception?

Reply:
It's possible that we might be able to, I didn't look too deep into where we use them, since I was mostly focusing on type-aliases obviously.
But I suspect that we might use these for other purposes, like for error reporting, in cases where just using the name (and not the full type-id) might not be descriptive enough. I can look into it though and see!